odxtools/diaglayers/protocol.py

```diff
@@ -24,14 +26,26 @@ class Protocol(HierarchyElement):
     def protocol_raw(self) -> ProtocolRaw:
         return cast(ProtocolRaw, self.diag_layer_raw)
 
+    @property
+    def comparam_spec_ref(self) -> OdxLinkRef:
+        return self.protocol_raw.comparam_spec_ref
+
     @property
     def comparam_spec(self) -> ComparamSpec:
         return self.protocol_raw.comparam_spec
 
+    @property
+    def prot_stack_snref(self) -> str | None:
+        return self.protocol_raw.prot_stack_snref
```

Comment:
Really needed after the database links has been resolved?

Reply:
logic: provide the user with the stack itself, not just its ref

Reply:
why not have both? That way, raw and "full" diag layers can be used interchangeably (for code which does not use the `.diag_layer_raw` attribute)...

Reply:
because it's already available as `prot_stack.odx_id`

Reply:
the ODX id is not exactly equivalent to a reference, e.g., the reference may feature a different set of document fragments. Also, all other classes which do reference resolution still provide the references in addition to the resolved objects..